cmd/helm-operator/main.go

```diff
@@ -189,12 +191,12 @@ func main() {
 	go statusUpdater.Loop(shutdown, log.With(logger, "component", "annotator"))
 
 	gitRemote := git.Remote{URL: *gitURL}
-	repo := git.NewRepo(gitRemote, git.PollInterval(*gitPollInterval), git.ReadOnly)
+	repo := git.NewRepo(gitRemote, git.PollInterval(*gitPollInterval), git.TimeOut(*gitTimeOut), git.ReadOnly)
 
 	// 		Chart releases sync due to Custom Resources changes -------------------------------
 	{
 		mainLogger.Log("info", "Attempting to clone repo ...", "url", gitRemote.URL)
-		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Why did we drop the timeout here?

Reply:
Because every git operation (step) has its own time-out set. If we leave the time-out here it could cut off early instead of respecting the user’s configuration flag.